From 1c15312ad443a764eeecc4f5ebe26c6e1060423c Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Mon, 6 Jun 2016 18:08:46 +0800
Subject: [PATCH 0530/1722] drm/amd/dal: clean-up internal payloads table

Add code to clean-up internal MST payload table in case
MST upstream device got disconnected. There was an issue
that in case first mode set occurs on VCP ID which
already exist in table, old value from table was used,
which lead to intermittent issues with target enablement,
when either incorrect number of slots or stream encoder
where picked up

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index 2354813..432fe7d 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -745,6 +745,9 @@ bool dc_link_detect(const struct dc_link *dc_link, bool boot)
 			LINK_INFO("link=%d, mst branch is now Disconnected\n",
 				link->public.link_index);
 			dm_helpers_dp_mst_stop_top_mgr(link->ctx, &link->public);
+
+			link->mst_stream_alloc_table.stream_count = 0;
+			memset(link->mst_stream_alloc_table.stream_allocations, 0, sizeof(link->mst_stream_alloc_table.stream_allocations));
 		}
 
 		link->public.type = dc_connection_none;
-- 
2.7.4

